staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -54,6 +54,8 @@ type CustomResourceDefinitionSpec struct {
 	Versions []CustomResourceDefinitionVersion `json:"versions,omitempty" protobuf:"bytes,7,rep,name=versions"`
 	// AdditionalPrinterColumns are additional columns shown e.g. in kubectl next to the name. Defaults to a created-at column.
 	AdditionalPrinterColumns []CustomResourceColumnDefinition `json:"additionalPrinterColumns,omitempty" protobuf:"bytes,8,rep,name=additionalPrinterColumns"`
+	// Prune enables pruning of fields not specified in validation.openAPIV3Schema. Note: this will default to true in version v1.
+	Prune bool `json:"prune,omitempty" protobuf:"varint,9,opt,name=prune"`
```

Comment:
Shouldn't this be an optional value (`*bool`)?

Reply:
It defaults to false in v1beta1, to true in newly created GA CRDs. So technically we can keep it a non-pointer now.

Reply:
I was more concerned with the api-conventions doc which I linked above

Reply:
If you need to round trip a false value (from a CRD created in v1beta1) in new versions of CRDs, you need `*bool`

Reply:
To agree about the plan, this it is (if we go with `bool`):

1. an object being created via v1beta1 we default to false.
2. an old v1beta1 object read from etcd is defaulted to false.
3. v1 and internal will get a bool pointer
4. an v1 object defaults to `*true` when created (and read from storage, but in storage we know to have it set anyway).
5. when converting from v1beta1 to internal `false` convert to `*false`, `true` to `*true`.
6. when converting from internal to v1beta1, nil never exists (but we can convert to true, `*false` defaults to false, `*true` default to `true`.
7. we keep the omitempty in v1beta1.

Instead, we could use `*bool` in v1 already. It makes the transition easier maybe. @liggitt opinion? you have had fights with similar cases before. Am happy with either way.

Reply:
> Instead, we could use *bool in v1 already.

I think it'll be easier to have `*bool` in both `internal`, `v1` and `v1beta1` making the conversion without a problem and it'll ease developers who develop against `v1beta1` and want to switch to `v1` once its released

in `v1beta1/zz_generated.defaults.go` we might have:

```go
defaultPrune := false
if in.Spec.Prune == nil {
     in.Spec.Prune = &defaultPrune
}
```

and in `v1/zz_generated.defaults.go`:

```go
defaultPrune := true
if in.Spec.Prune == nil {
     in.Spec.Prune = &defaultPrune
}
```

Reply:
at least it is symmetric. I will change it to `*bool` for v1beta1.

Reply:
done